client/src/app/dashboard/page.tsx

```diff
@@ -33,88 +52,285 @@ const PasswordPage = () => {
-        (obj: any) => obj._id !== newPassword._id
-      );
-
-      getPasswordDeleteAPI(
-        () => () => PasswordsService.DeleteUserPassword(newPassword._id, token)
-      );
+      setPasswordsData(updatedPasswords);
+      setShowPopUp((prev) => ({ ...prev, status: false }));
     }
+  }, [editData, isEditDataError]);
 
+  const onDeletePassword = (id: string) => {
+    getPasswordDeleteAPI(() => () => PasswordsService.DeleteUserPassword(id, token));
+    const updatedPasswords = passwordsData.filter((obj: any) => obj._id !== id);
     setPasswordsData(updatedPasswords);
+    setShowPopUp((prev) => ({ ...prev, status: false }));
+  };
+
+  const handleSubmit = (type: string, item: any) => {
+    if (type === "add") {
+      getPasswordAddAPI(() => () => PasswordsService.addUserPassword(item, token));
+    } else {
+      getPasswordEditAPI(() => () => PasswordsService.UpdateUserPassword(item._id, item, token));
+    }
+  };
+
+  const [isExpanded, setIsExpanded] = useState<boolean[]>(Array(passwordsData.length).fill(false));
+  const [passwords, setPasswords] = useState<string[]>(passwordsData.map((item: any, index: number) => item.password));
+  const [copiedIndex, setCopiedIndex] = useState<number | null>(null);
+
+  const toggleExpand = (index: number) => {
+    setIsExpanded(prevState => {
+      const newExpandedCards = [...prevState];
+      newExpandedCards[index] = !newExpandedCards[index];
+      return newExpandedCards;
+    });
+  };
+
+  const handlePasswordChange = (index: number, e: React.ChangeEvent<HTMLInputElement>) => {
+    const newPasswords = [...passwords];
+    newPasswords[index] = e.target.value;
+    setPasswords(newPasswords);
   };
 
+  const handleCopy = async (index: number) => {
+    try {
+      await navigator.clipboard.writeText(passwords[index]);
+      setCopiedIndex(index);
+      setTimeout(() => setCopiedIndex(null), 3000);
+    } catch (err) {
+      console.error("Failed to copy: ", err);
+    }
+  };
+
+  // Split the data into two columns
+  const column1Data = passwordsData.filter((item: any, index: number) => index % 2 === 0);
+  const column2Data = passwordsData.filter((item: any, index: number) => index % 2 !== 0);
+
   return (
     <AuthHoc>
-      <AuthHeader />
-      <div data-testid="password-page" className="">
-        <div className=" flex flex-row justify-between items-center px-2 py-4">
-          <h1 className="font-medium">Your Saved Passwords</h1>
-          <button
-            className="text-white bg-teal-500 p-1 rounded"
-            onClick={() => setShowPopUp({ type: "add", status: true })}
-          >
-            <p className="">+Add New</p>
-          </button>
-        </div>
-
-        {showPopUp.type === "add" && showPopUp.status ? (
-          <CreatePassword
-            onClose={() => setShowPopUp((prev) => ({ ...prev, status: false }))}
-            onSubmit={onAddPassword}
-          />
-        ) : null}
-
-        <div className="m-2 p-2 lg:w-1/2">
```

Comment:
Follow the same structure create a separate component for password item instead of adding in the same dashboard. Reason we need to break the page into several components as much as possible so that the code looks cleaner. That's the react came into picture as breaking the components. 
Also the reason of creating separate password item component is we don't need to iterate and match the item id for decrypt the password as it will increase time complexity too.

Follow the same: https://github.com/abhijeetnishal/Password-Shield/blob/ui-fix/client/src/app/dashboard/page.tsx structure